File: ironpythoninaction/chapter15/15.3/PluginEmbeddingVB/PluginEmbeddingVB/PluginStore.vb
Imports Plugins


Public Class PluginStore

    Private Shared _plugins As List(Of PluginBase) = New List(Of PluginBase)()

    Friend Shared ReadOnly Property Plugins() As List(Of PluginBase)
        Get
            Return _plugins
        End Get
    End Property

    Public Shared Sub AddPlugin(ByVal plugin As PluginBase)
        _plugins.Add(plugin)
    End Sub
End Class

File: ironpythoninaction/chapter15/15.3/EmbeddingPlugin/EmbeddingPlugin/PluginStore.cs
using System;
using System.Collections.Generic;
using System.Text;
using Plugins;
using System.Windows.Forms;

namespace EmbeddingPlugin
{

    public class PluginStore
    {
        private static List<PluginBase> _plugins = new List<PluginBase>();

        internal static List<PluginBase> Plugins
        {
            get { return _plugins; }
        }

        public static void AddPlugin(PluginBase plugin)
        { _plugins.Add(plugin); }

    }

}
